hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -1124,4 +1125,46 @@ public void forEach(@Nonnull BiConsumer<? super K, ? super V> action) {
         }
     }
 
+    public V compute(@Nonnull K key, @Nonnull BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(key, NULL_BIFUNCTION_IS_NOT_ALLOWED);
+
+        if (SerializationUtil.isClassStaticAndSerializable(remappingFunction)
+                && isClusterVersionGreaterOrEqual(Versions.V4_1)) {
+            ComputeEntryProcessor<K, V> ep = new ComputeEntryProcessor<>(remappingFunction);
+            return executeOnKey(key, ep);
+        } else {
+            return computeLocally(key, remappingFunction);
+        }
+    }
+
+    private V computeLocally(K key,
+                                      BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+
+        while (true) {
```

Comment:
Unbound looping.

Reply:
Same comment - not sure how we can replace the value without retrying indefinitely. That's always going to be a problem if the mapping function is not serializable.